### PR TITLE
Fix sentence breaker matching

### LINKED ISSUE #1210: Passive sentence breaker matching is broken
## What did you expect to happen?
The sentence part breakers only to break the sentence into parts when the breaker is a word on its own. For example, it should break on `is` but not on `pra*is*e`.

## What happened instead?
It did break on praise.

## How can we reproduce this behavior?
- Use the browserified example. Don't forget to run a `grunt build:js`.
- Add `console.log(sentencePart)` to `js/researches/english/determinePassives`
- Write `Who commented is praise due`.
- The logged sentence part is `is pra`

### PR DESCRIPTION
**Pull request by manuelaugustin**
## Summary

This PR can be summarized in the following changelog entry: Fixes a bug where the passive voice detection was broken because of incorrect sentence breaker matching.

## Test instructions

This PR can be tested by following these steps:

- Use the browserified example. Don't forget to run a `grunt build:js`.
- Add `console.log(sentencePart)` to `js/researches/english/determinePassives`
- Write `Who commented is praise due`.
- The logged sentence part is `is pra`

- Check out this branch.
- Don't forget to run a `grunt build:js`.
- Add the console.log for the sentencePart.
- The logged sentence part should be `is praise due`.

Fixes #1210

### CODE REVIEW

**Comment by IreneStr** [comment]
Travis error is unrelated to this PR and is fixed in https://github.com/Yoast/YoastSEO.js/pull/1212

File: spec/stringProcessing/indicesSpec.js [hunk 1]
```diff
@@ -92,6 +93,20 @@ describe( "A function to filter out overlapping matches", function() {
 
 });
 
+describe( "A function that returns all instances of the input word and their index", function() {
```
Comment (Xyfi) [inline spec/stringProcessing/indicesSpec.js]: I would add tests for cases where no matches are found and for multiple matches. This seems like important functionality for this function. Also tests where you pass invalid or `null` parameters to `getIndicesByWord` could be a good idea.

Comment (Xyfi) [inline spec/stringProcessing/indicesSpec.js]: Another comment on the tests file in general: How do I know what function is tested by a test specifically, since only a description is given?

File: js/stringProcessing/indices.js [hunk 2]
```diff
@@ -15,12 +16,19 @@ function getIndicesByWord( word, text ) {
...
 	var index, indices = [];
 	while ( ( index = text.indexOf( word, startIndex ) ) > -1 ) {
-		indices.push(
-			{
-				index: index,
-				match: word,
-			}
-		);
+		// Check if the previous and next character are word boundaries to determine if a complete word was detected
+		var previousCharacter = characterInBoundary( text[ index - 1 ] ) || index === 0;
```
Comment (Xyfi) [inline js/stringProcessing/indices.js]: `isPreviousCharacterBoundary` and `isNextCharacterBoundary` seem better variable names.

**Comment by Xyfi** [comment]
CR 🚧

File: spec/stringProcessing/indicesSpec.js (hunk 1 above)
Comment (IreneStr) [inline spec/stringProcessing/indicesSpec.js]: In the `expect(...)` you can see which function in tested

**Comment by boblinthorst** [comment]
CR; seems fine after removing Irene's accidental commits (move cat... && add func....).
